keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyAsyncClient.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    public Mono<Response<Key>> getKey(String name) {
+        return getKey(name, "");
+    }
+
+
+    public Mono<Response<Key>> getKey(KeyBase keyBase) {
+        Objects.requireNonNull(keyBase, "The Key Base parameter cannot be null.");
+        String keyVersion = "";
+
+        return service.getKey(endpoint, keyBase.name(), keyVersion, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+    public Mono<Response<Key>> updateKey(Key key) {
+        Objects.requireNonNull(key, "The key input parameter cannot be null.");
+        KeyRequestParameters parameters = new KeyRequestParameters()
+                .tags(key.tags())
+                .keyOps(key.keyOperations())
+                .keyAttributes(new KeyRequestAttributes(key));
+
+        return service.updateKey(endpoint, key.name(), key.version(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Mono<Response<DeletedKey>> deleteKey(String name) {
+        return service.deleteKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Mono<Response<DeletedKey>> getDeletedKey(String name) {
+        return service.getDeletedKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Mono<VoidResponse> purgeDeletedKey(String name) {
+        return service.purgeDeletedKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Mono<Response<Key>> recoverDeletedKey(String name) {
+        return service.recoverDeletedKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Mono<Response<byte[]>> backupKey(String name) {
+        return service.backupKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE)
+                .flatMap(base64URLResponse ->  Mono.just(new SimpleResponse<byte[]>(base64URLResponse.request(),
+                base64URLResponse.statusCode(), base64URLResponse.headers(), base64URLResponse.value().value())));
+    }
+
+
+    public Mono<Response<Key>> restoreKey(byte[] backup) {
+        KeyRestoreRequestParameters parameters = new KeyRestoreRequestParameters().keyBackup(backup);
+        return service.restoreKey(endpoint, API_VERSION, parameters, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE);
+    }
+
+
+    public Flux<KeyBase> listKeys() {
+        return service.getKeys(endpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(this::extractAndFetchKeys);
+    }
+
+
+    public Flux<DeletedKey> listDeletedKeys() {
+        return service.getDeletedKeys(endpoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE).flatMapMany(this::extractAndFetchDeletedKeys);
+    }
+
+
+    public Flux<KeyBase> listKeytVersions(String name) {
```

Comment:
looks like a typo but not sure, it says `listKeytVersions`

Reply:
updated